lib/AST/Module.cpp

```diff
@@ -2182,12 +2203,15 @@ ModuleDecl::computeMagicFileStringMap(bool shouldDiagnose) const {
 SourceFile::SourceFile(ModuleDecl &M, SourceFileKind K,
                        Optional<unsigned> bufferID,
                        bool KeepParsedTokens, bool BuildSyntaxTree,
```

Comment:
A bit of cleanup would be to hand SourceFile some `LangOptions`. We're now up to 3 bits and an option set here.

Reply:
@CodaFi I'm planning on sinking `KeepParsedTokens` & `BuildSyntaxTree` down into `ParsingOptions` in a future PR, how does that sound?

Reply:
That'll do just fine.